scripts/codemods/explicit-barrel-imports.cjs

```diff
@@ -1,39 +1,43 @@
-const fs = require('fs');
 const path = require('path');
 
+const isBareSpecifier = (importPath) => !importPath.startsWith('app/') && /^[^./]/.test(importPath);
+const barrelFileNames = ['index.ts', 'index.tsx', 'index.js', 'index.jsx'];
+
+const resolvePath = (fileDir, importPath) => {
+  if (importPath.startsWith('app/')) {
+    const resolvedPath = require.resolve(path.join(process.cwd(), 'public', importPath));
+    return resolvedPath;
+  }
+
+  return require.resolve(path.resolve(fileDir, importPath));
+};
+
 module.exports = function (fileInfo, api) {
   const j = api.jscodeshift;
   const root = j.withParser('tsx')(fileInfo.source);
   const fileDir = path.dirname(fileInfo.path);
-
-  // Function to check if the path potentially points to a barrel file
-  const mightBeBarrelFileImport = (importPath) => {
-    const fullPath = path.join(fileDir, importPath);
-    if (fs.existsSync(fullPath) && fs.lstatSync(fullPath).isDirectory()) {
-      if (fs.existsSync(path.join(fullPath, 'index.ts')) || fs.existsSync(path.join(fullPath, 'index.js'))) {
-        return true;
-      }
-    }
-    return false;
-  };
-
-  // Udpate import declarations that import from barrel files
+  // Update import declarations that import from barrel files
   root
     .find(j.ImportDeclaration)
-    .filter((path) => mightBeBarrelFileImport(path.node.source.value))
+    .filter((path) => !isBareSpecifier(path.node.source.value))
     .forEach((path) => {
-      // Create a comment node
-      const comment = j.commentLine(' @todo: replace barrel import path');
+      const resolvedPath = resolvePath(fileDir, path.node.source.value);
+      if (barrelFileNames.some((barrelFileName) => resolvedPath.endsWith(barrelFileName))) {
```

Comment:
Nit
```suggestion
      if (barrelFileNames.some(resolvedPath.endsWith)) {
```

Reply:
I'm not sure we want to do this. Won't this mean we pass the index of the current item in the array as the endPosition arg of `endsWith`?

Reply:
Ohh, that's true, pardon!